samples/bluetooth/hci_spi/src/main.c

```diff
@@ -0,0 +1,288 @@
+			/* Unblock the spi tx thread and wait for it */
+			k_sem_give(&sem_spi_tx);
+			k_sem_take(&sem_spi_rx, K_FOREVER);
+			continue;
+		}
+
+		/* Receiving data from the SPI Host */
+		ret = spi_transceive(spi_hci_dev, &txmsg, SPI_MAX_MSG_LEN,
+				     &rxmsg, SPI_MAX_MSG_LEN);
+		if (ret < 0) {
+			SYS_LOG_ERR("SPI transceive error: %d", ret);
+		}
+
+		switch (rxmsg[PACKET_TYPE]) {
+		case HCI_CMD:
+			memcpy(&cmd_hdr, &rxmsg[1], sizeof(cmd_hdr));
+
+			buf = net_buf_alloc(&cmd_tx_pool, K_NO_WAIT);
+			if (buf) {
+				bt_buf_set_type(buf, BT_BUF_CMD);
+				net_buf_add_mem(buf, &cmd_hdr,
+						sizeof(cmd_hdr));
+				net_buf_add_mem(buf, &rxmsg[4],
+						cmd_hdr.param_len);
+			} else {
+				SYS_LOG_ERR("No available command buffers!");
+				continue;
+			}
+			break;
+		case HCI_ACL:
+			memcpy(&acl_hdr, &rxmsg[1], sizeof(acl_hdr));
+
+			buf = net_buf_alloc(&acl_tx_pool, K_NO_WAIT);
+			if (buf) {
+				bt_buf_set_type(buf, BT_BUF_ACL_OUT);
+				net_buf_add_mem(buf, &acl_hdr,
+						sizeof(acl_hdr));
+				net_buf_add_mem(buf, &rxmsg[5],
+						sys_le16_to_cpu(acl_hdr.len));
+			} else {
+				SYS_LOG_ERR("No available ACL buffers!");
+				continue;
+			}
+			break;
+		default:
+			SYS_LOG_ERR("Unknown BT HCI buf type");
+			continue;
+		}
+
+		SYS_LOG_DBG("buf %p type %u len %u",
+			    buf, bt_buf_get_type(buf), buf->len);
+
+		bt_send(buf);
+		stack_analyze("tx_stack", bt_tx_thread_stack,
+			      sizeof(bt_tx_thread_stack));
+
+		/* Make sure other threads get a chance to run */
+		k_yield();
+	}
+}
+
+static int hci_spi_init(struct device *unused)
+{
+	static struct spi_config btspi_config = {
+		.config = SPI_WORD(8),
+	};
+
+	SYS_LOG_DBG("");
+
+	spi_hci_dev =
+		device_get_binding(CONFIG_BLUETOOTH_SPI_TO_HOST_DEV_NAME);
+	if (!spi_hci_dev) {
+		return -EINVAL;
+	}
+
+	if (spi_configure(spi_hci_dev, &btspi_config) < 0) {
+		return -EINVAL;
+	}
+
+	gpio_dev = device_get_binding(GPIO_IRQ_DEV_NAME);
+	if (!gpio_dev) {
+		return -EINVAL;
+	}
+	gpio_pin_configure(gpio_dev, GPIO_IRQ_PIN,
+			   GPIO_DIR_OUT | GPIO_PUD_PULL_DOWN);
+
+	return 0;
+}
+
+DEVICE_INIT(hci_spi, "hci_spi", &hci_spi_init, NULL, NULL,
+	    APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
+void main(void)
+{
+	static K_FIFO_DEFINE(rx_queue);
+	struct bt_hci_evt_hdr *evt_hdr;
+	struct net_buf *buf;
+	int err;
+
+	SYS_LOG_DBG("Start");
+
+	bt_enable_raw(&rx_queue);
+
+	/* Spawn the TX thread, which feeds cmds and data to the controller */
+	k_thread_create(&bt_tx_thread_data, bt_tx_thread_stack,
+			sizeof(bt_tx_thread_stack), bt_tx_thread,
+			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
+
+	/* Send a vendor event to announce that the slave is initialized */
+	buf = net_buf_alloc(&cmd_tx_pool, K_FOREVER);
+	if (buf) {
+		bt_buf_set_type(buf, BT_BUF_EVT);
+		evt_hdr = net_buf_add(buf, sizeof(*evt_hdr));
+		evt_hdr->evt = BT_HCI_EVT_VENDOR;
+		evt_hdr->len = 2;
+		u16_t *param = net_buf_add(buf, sizeof(u16_t));
+		*param = sys_cpu_to_le16(EVT_BLUE_INITIALIZED);
```

Comment:
This looks like a potential unaligned access violation since you can't be 100% sure that the pointer returned by net_buf_add sits on a 2-byte boundary. You should instead use net_buf_add_le16(buf, EVT_BLUE_INITIALIZED).